rotkehlchen/rotkehlchen.py

```diff
@@ -1050,17 +1050,31 @@ def query_balances(
                 )
             else:
                 if len(nft_balances) != 0:
-                    if str(Location.BLOCKCHAIN) not in balances:
+                    if (blockchain_location := str(Location.BLOCKCHAIN)) not in balances:
                         balances[str(Location.BLOCKCHAIN)] = {}
 
                     for balance_entry in nft_balances:
                         if balance_entry['usd_price'] == ZERO:
                             continue
-                        balances[str(Location.BLOCKCHAIN)][CryptoAsset(
-                            balance_entry['id'])] = Balance(
-                            amount=ONE,
-                            usd_value=balance_entry['usd_price'],
-                        )
+
+                        # It can happen that the asset was manually added
+                        # as a token and we don't want to ignore NFTs from the token query since
+                        # they might not be tracked by Opensea. In case of them being already
+                        # in the chain balances we update the price and continue
+                        blockchain_balances = balances[blockchain_location]
+                        nft = Nft(balance_entry['id'])
+                        nft_as_token = GlobalDBHandler.get_evm_token(
+                            address=nft.evm_address,
+                            chain_id=nft.chain_id,
+                        )  # we need the eip155 identifier instead of the _nft_ one.
+
+                        if nft_as_token in blockchain_balances:
+                            blockchain_balances[nft_as_token].usd_value = balance_entry['usd_price']  # noqa: E501
```

Comment:
Why update the price here? They had no price before? What if previous price was more correct?